components/channel_select/channel_select.tsx

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+
+import Constants from 'utils/constants';
+import * as Utils from 'utils/utils.jsx';
+
+type Props = {
+    channels: Array<{
```

Comment:
We have a type for `Channel` in `mattermost-redux` that you can use here instead.

Reply:
Agreed.

Reply:
Sure thing, I'll make the change.